### DISCUSSION TITLE
Feedback

### DISCUSSION THREAD

**Discussion by QTDerg** [QTDerg/switcher-roo#4] [General]
Post your feedback, opinions, criticism or feature requests here!

**Comment by SRThrowaway** [comment]
The Android APK is broken. It only shows "Fatal error" now. I've submitted a PR to replace the APK with a PWA, which is easier to maintain and has higher compatibility with non Android devices. I might have missed a few other changes I made while testing, as I had to edit some things to work with GitHub Pages, so check my master branch if anything seems wrong.

**Comment by QTDerg** [comment]
Sorry for late reply, didn't checked github for a while.

There is one problem I'm using InfinityFree and free SSL certificate (from Let's Encrypt) doesn't work with their free subdomains. https://infinityfree.net/support/cannot-use-or-have-problems-using-lets-encrypt-or-sslforfreecom/

Btw, what device and android version you are on? APK is working fine on my S6 with Android 7.0

I also noticed you couldn't download .apk from the GDrive link as the Google thinks it's a virus lol. I used Website 2 APK Builder to make the APK and after quick googling I found out others have this problem too https://websitetoapk.com/forums/100-anti-virus-reports-the-app-created-is-malware/0

**Comment by SRThrowaway** [comment]
I'm using a OnePlus 3T with Android 10. 

I would change that domain to redirect to a GitHub Pages hosted instance then. HTTPS is the default setting, and there is no reason why any site should be http only in 2020.

**Comment by QTDerg** [comment]
Ok, I will switch to hosting site directly from Github then. The folks will lose their saved data though (I'm using Localstorage for it), but I will leave rf.gd running for a while so they can transfer it.

**Comment by noah-afk** [comment]
What is the secret admin code, I would love to see what it does

**Comment by QTDerg** [comment]
Admin password is "OwO" (without quotation marks)

**Comment by noah-afk** [comment]
And what does it do

**Comment by Hexanet1226** [comment]
Asking here if it would be ok to work on an Apk file, I can do that for myself and give the file to be posted, I know some people that would have it better as an APK and I can work on making it work better as one if given permission to

**Comment by QTDerg** [comment]
Yeah, you can make APK version. You have my permission.

**Comment by coolguy19** [comment]
so this isnt a problem with the game but could you add something perhaps for those of us who have terrible memory? I made a lock password in the app and almost immediately forgot it now I am stuck on an unsupported species

**Comment by coolguy19** [comment]
> so this isnt a problem with the game but could you add something perhaps for those of us who have terrible memory? I made a lock password in the app and almost immediately forgot it now I am stuck on an unsupported species

like maybe add a button that sends a link to your email if you cant remember the password or when you make a password lock you get send an email with that password lock?

**Comment by QTDerg** [comment]
- Go to Options
- Tick "Show Developer Menu button"
- Go to Developer Menu
- Tap "Delete Lock password"

I'm thinking of a way to let folks reset their password if they forgot it. The app is 100% client-side so e-mails are out of question since a server is needed to send an e-mail. Maybe "I forgot my password" button in Lock password menu that let's you reset it?

**Comment by coolguy19** [comment]
> * Go to Options
> * Tick "Show Developer Menu button"
> * Go to Developer Menu
> * Tap "Delete Lock password"
> 
> I'm thinking of a way to let folks reset their password if they forgot it. The app is 100% client-side so e-mails are out of question since a server is needed to send an e-mail. Maybe "I forgot my password" button in Lock password menu that let's you reset it?

perfect. something for those of us who leave the app for a while or just have bad memory

**Comment by coolguy19** [comment]
> * Go to Options
> * Tick "Show Developer Menu button"
> * Go to Developer Menu
> * Tap "Delete Lock password"
> 
> I'm thinking of a way to let folks reset their password if they forgot it. The app is 100% client-side so e-mails are out of question since a server is needed to send an e-mail. Maybe "I forgot my password" button in Lock password menu that let's you reset it?

that fixed the lock now it wont stop crashing when I load it in...

**Comment by coolguy19** [comment]
I even restarted my phone and it just keeps crashing every time I try to load it in

**Comment by coolguy19** [comment]
I have the feeling creative mode with the addition of character viz enabled might be the culprit

**Comment by QTDerg** [comment]
That's bad :c I'm sorry to hear you're experiencing issues

CharViz can be quite performance intensive feature for phones. Though my phone is a bit dated (SGS6) I was able scroll down to turn it off in options when I was testing it. It lagged for sure but didn't crash

I think I will make it so you need to tap the CharViz window if you want visualizer to load. That way people won't be locked out of the app if their device can't handle it. Of course I will add an option to load it automatically so people don't have to do it everytime

**Comment by coolguy19** [comment]
> That's bad :c I'm sorry to hear you're experiencing issues
> 
> CharViz can be quite performance intensive feature for phones. Though my phone is a bit dated (SGS6) I was able scroll down to turn it off in options when I was testing it. It lagged for sure but didn't crash
> 
> I think I will make it so you need to tap the CharViz window if you want visualizer to load. That way people won't be locked out of the app if their device can't handle it. Of course I will add an option to load it automatically so people don't have to do it everytime

yah it's just character viz. for registered characters. whenever I load it in it instantly crashes the app

**Comment by coolguy19** [comment]
it is very rare that it doesnt crash

**Comment by coolguy19** [comment]
> That's bad :c I'm sorry to hear you're experiencing issues
> 
> CharViz can be quite performance intensive feature for phones. Though my phone is a bit dated (SGS6) I was able scroll down to turn it off in options when I was testing it. It lagged for sure but didn't crash
> 
> I think I will make it so you need to tap the CharViz window if you want visualizer to load. That way people won't be locked out of the app if their device can't handle it. Of course I will add an option to load it automatically so people don't have to do it everytime

have you tried using something less intense for character viz?

**Comment by QTDerg** [comment]
I will try to optimize CharViz to be less resource intensive

Meanwhile I just pushed an update addressing your issue. Now you need to tap the visualizer window to load CharViz so app shouldn't crash on load anymore
![charviz launch](https://user-images.githubusercontent.com/51024661/126083708-b0595e10-3c1a-4cd7-ba68-7b7bdb3fa721.png)

**Comment by coolguy19** [comment]
> I will try to optimize CharViz to be less resource intensive
> 
> Meanwhile I just pushed an update addressing your issue. Now you need to tap the visualizer window to load CharViz so app shouldn't crash on load anymore
> ![charviz launch](https://user-images.githubusercontent.com/51024661/126083708-b0595e10-3c1a-4cd7-ba68-7b7bdb3fa721.png)

after the less intense thing you find for character viz, maybe work on the additional settings like what the characters will look like when pregnant or have low intelligence. and show the genetalia

**Comment by coolguy19** [comment]
> > I will try to optimize CharViz to be less resource intensive
> > Meanwhile I just pushed an update addressing your issue. Now you need to tap the visualizer window to load CharViz so app shouldn't crash on load anymore
> > ![charviz launch](https://user-images.githubusercontent.com/51024661/126083708-b0595e10-3c1a-4cd7-ba68-7b7bdb3fa721.png)
> 
> after the less intense thing you find for character viz, maybe work on the additional settings like what the characters will look like when pregnant or have low intelligence. and show the genetalia

or even add more characters

**Comment by coolguy19** [comment]
all I have are ideas. if I had anything to help then I would use a more 3d program

**Comment by coolguy19** [comment]
I think the most important thing before adding more characters or outfits or slider interactions with the characters we make, a program needs to be found that is less impactfull. then I think adding what they look like with the different intelligence slider positions would be hot

**Comment by Caldoric** [comment]
> > I will try to optimize CharViz to be less resource intensive
> > Meanwhile I just pushed an update addressing your issue. Now you need to tap the visualizer window to load CharViz so app shouldn't crash on load anymore
> > ![charviz launch](https://user-images.githubusercontent.com/51024661/126083708-b0595e10-3c1a-4cd7-ba68-7b7bdb3fa721.png)
> 
> after the less intense thing you find for character viz, maybe work on the additional settings like what the characters will look like when pregnant or have low intelligence. and show the genetalia

Yeah, all it currently does is either (when 18+ is enabled) show a dick/boobs based on gender toggle(s), or (when 18+ is disabled) not show anything.

**Comment by Dthrowaway** [comment]
Found an issue a bit ago, randomizing doesn't properly update the opposite side while connected to someone, tested locally and with someone over a proper connection.

**Comment by QTDerg** [comment]
@Dthrowaway To be honest I never intended Randomizer to work with the whole Control thing but it should work now. Only Master can use the Randomizer.

@Caldoric Here's list of what CharViz does for now:
- Thin/Thicc slider (affects body type, 5 keyframes)
- Breast Size slider (flat + 4 sizes)
- Hair Lenght slider (no hair, short, long)
- Cock Type option (only Humanoid type available)
- Genital Size slider (affects dick size, 5 keyframes)
- Overall 25 species supported (Dragon, Shark + other scalies and canines). You can also make a hybrid character using Part Picker in CharViz menu.

I know it's a bit lackluster but bringing it to state that everyone would be satisfied of would be enormous amount of work and I just have no motivation to do it. Maybe I will add some clothes, support for female genitalia and add a few dick types but that'll be it.
![randomizerControl](https://user-images.githubusercontent.com/51024661/136294698-9e80c1af-f6e9-4350-8aab-4c870a1e2459.png)

**Comment by Dthrowaway** [comment]
@QTDerg Fantastic work overall!

I ended up checking on the presets as well, this time using only local connections, though that seems to be consistent regardless, loading presets don't accurately reflect it on the opposite end either, similar to the randomzier issue. It updates the values of the category (Sliders, Checkboxes, Radio Buttons, etc.) but whole give it a proper label until something in the category is 'Set'.

**Comment by QTDerg** [comment]
@Dthrowaway 

I just pushed an update. Loading presets should be synced now :3

**Comment by Siveen** [comment]
I've looked around a lil and haven't found anything, maybe a lil' troubleshooting guide somewhere in the app would be cool. 
In my case I'd have to manually open ports, but don't know which ones.

**Comment by QTDerg** [comment]
@Siveen 

Could you check if it works for you now? I swapped a TURN server to (hopefully) working one from Open Relay

TURN server acts as relay between the users when direct connection cannot be established among them. You can read about it here -> https://www.metered.ca/tools/openrelay/

If it doesn't help, you could try opening ports 80, 443, 3478, 5004. App uses WebRTC for communication between users

**Comment by shelbyKiraM** [comment]
Loving the tool; using on iOS 16 beta 2, works as expected. I'ma dev, just wonderin'... How does the server sync work? Can I host a page on my own server to store the data and sync between browser & home screen bookmark versions? Plus like maybe username and password? Or maybe an export/import or copy/paste button of base64 settings of the saved characters? I can fork and submit PRs to maybe do some of that after thxgivin' ends. Fun fun!

😽😁😋🏳️‍🌈💖

**Comment by Shadowcre** [comment]
This is my first time using it, and I've just realized that it's real after quite a while. During my first attempt to use it, I encountered a problem. After trying to perform its function, it took a little bit of loading time. I thought it might take too long, so I decided to exit the app, hoping it would continue in the background. However, upon returning, I found that it had reset. I decided to try again, adding extra bits, but it still won't work at all. I don’t know what’s wrong with it. I'm unsure if it's an actual bug with the app or if I really did mess something up.

(A little edit before posting): I just realized that I figured out the problem, and I now understand that I just need a little time off from using it. This allows me to wait a while, while the changes to the character I started making generate or whatever. It's a little bit by itself individually. So I guess I didn’t need to get off the app and remove it from recently played apps. But I do think this is actually quite wonderful. And I think there could be another feature for it that I’m now thinking of suggesting. It doesn’t have to be too complicated with its function, but maybe a feature where different users can add pictures of their original characters, and it could then generate a rough depiction of them. That’s not too complicated, but can be simplistic enough for them to make manual changes that fit them best.

**Comment by zankermichael1995-star** [comment]
Is there a more recent version of the app? If so, where can I find it?

**Reply by volcano2207** [reply]
wondering the same thing

**Comment by volcano2207** [comment]
is the dev charviz stuff ever going to be finished? or is this abanoded ware?
